source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -95,6 +211,12 @@ void Cluster::onDnsHostAddOrUpdate(
   swapAndUpdateMap(new_host_map, hosts_added, {});
 }
 
+bool Cluster::createCustomTlsForHost() {
+  // TODO(mattklein123): Consider custom settings per host and/or global cluster config to turn this
```

Comment:
+1 - feels like setting transport socket for forward proxy should be considered a config error

Reply:
No this should be temporarily, transport socket should be freely set here, i.e. a TAP wrapping a TLS, or an alternative TLS impl (such as OpenSSL).

Reply:
Sorry what do you mean exactly? Are you saying that any overrides we do should be on the transport socket itself via options so that it works generically? If so I will update the TODO. I don't think I want to tackle that in this PR.

Reply:
> Sorry what do you mean exactly? Are you saying that any overrides we do should be on the transport socket itself via options so that it works generically? If so I will update the TODO. I don't think I want to tackle that in this PR.

Yeah, a TODO is fine here.